src/app/views/teams/teams-view/TeamsList.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { Link } from "react-router";
+import url from "../../../utilities/url";
+import Input from "../../../components/Input";
+import SimpleSelectableList from "../../../components/simple-selectable-list/SimpleSelectableList";
+
+const propTypes = {
+    dispatch: PropTypes.func,
+    teams: PropTypes.object.isRequired,
+    loadTeams: PropTypes.func.isRequired,
+};
+
+const TeamsList = props => {
+    const { teams, loadTeams } = props;
+
+    const [isFetchingTeams, setIsFetchingTeams] = useState(true);
+    const [filterTerm, setFilterTerm] = useState("");
+    const [previewTeams, setPreviewTeams] = useState([]);
+
+    useEffect(() => {
+        loadTeams();
+    }, []);
+
+    useEffect(() => {
+        const listOfPreviewTeams = [];
+        teams?.groups?.forEach(team => {
+            if (team.group_name.startsWith("role-")){
```

Comment:
it would be good to move this to selectors a that way it will be memorised
so create new selector for example getRoleGroups based on  getGroups  but with that logic inside.

Reply:
This is a really good shout and a good suggestion - was moving it but then after this morning discussion with TL  I have added 'role' type groups back into the list of preview teams so this filter no longer exists or is needed here.